src/App.tsx

```diff
@@ -14,11 +15,116 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selectedGoods: string[],
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  buttonAction = (good: string): void => {
+    if (!this.state.selectedGoods.includes(good)) {
+      this.setState((state) => {
+        state.selectedGoods.push(good);
+
+        return { selectedGoods: state.selectedGoods };
+      });
+    } else {
+      this.setState((state) => {
+        const newList = state.selectedGoods.filter(item => item !== good);
+
+        return { selectedGoods: newList };
+      });
+    }
+  };
+
+  messageLine = (list: string[] | []) => {
+    if (list.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (list.length === 1) {
+      return `${list.join(', ')} is selected`;
```

Comment:
you don't need to use `.join` here, you can find your only element from array by index `[0]`